data/json/terrain.json

```diff
@@ -5206,5 +5225,215 @@
+                { "item": "hose", "amount": 1},
+                { "item": "steel_chunk", "amount": 3, "minamount": 1 },
+                { "item": "bearing", "amount": 5, "minamount": 1 },
+                { "item": "frame", "amount": 1, "chance": 30},
+                { "item": "motor", "amount": 1, "chance": 30},
+                { "item": "splinter", "amount": 3, "chance": 30},
+                { "item": "2x4", "amount": 4, "minamount": 1 },
+                { "item": "nail", "amount": 10, "minamount": 3 }
+            ]
+        },
+        "bash": {
+            "str_min": 10, "str_max": 80,
+            "sound": "clang!",
+            "sound_fail": "ting.",
+            "ter_set": "t_floor",
+            "items": [
+                { "item": "wire", "amount": 1},
+                { "item": "pipe", "amount": 2, "minamount": 1 },
+                { "item": "chain", "amount": 1, "chance": 20},
+                { "item": "cu_pipe", "amount": 1, "chance": 10},
+                { "item": "scrap", "amount": 5, "minamount": 1 },
+                { "item": "steel_chunk", "amount": 2, "minamount": 1 },
+                { "item": "motor", "amount": 1, "chance": 10},
+                { "item": "splinter", "amount": 8, "minamount": 4 },
+                { "item": "2x4", "amount": 2},
+                { "item": "nail", "amount": 5, "minamount": 2 }
+            ]
+        }
+    },{
+        "type" : "terrain",
+        "id" : "t_machinery_electronic",
+        "name": "electronic machinery",
+        "symbol": "$",
+        "color": "yellow",
+        "move_cost": 8,
+        "flags": ["TRANSPARENT", "BASHABLE", "CONTAINER", "SEALED", "FLAMMABLE", "PLACE_ITEM"],
+        "deconstruct": {
+            "ter_set": "t_floor",
+            "items": [
+                { "item": "wire", "amount": 3, "minamount": 1 },
+                { "item": "pipe", "amount": 2, "minamount": 1 },
+                { "item": "steel_chunk", "amount": 4, "minamount": 1 },
+                { "item": "bearing", "amount": 6, "minamount": 2 },
+                { "item": "motor", "amount": 1, "chance": 40},
+                { "item": "processor", "amount": 1},
+                { "item": "RAM", "amount": 4, "minamount": 1 },
+                { "item": "cable", "amount": 4, "minamount": 1 },
+                { "item": "small_lcd_screen", "amount": 1},
+                { "item": "e_scrap", "amount": 10, "minamount": 5 },
+                { "item": "circuit", "amount": 8, "minamount": 3 },
+                { "item": "power_supply", "amount": 3, "minamount": 1 },
+                { "item": "amplifier", "amount": 3, "minamount": 1 },
+                { "item": "plastic_chunk", "amount": 8, "minamount": 2 },
+                { "item": "scrap", "amount": 5, "minamount": 1 }
+            ]
+        },
+        "bash": {
+            "str_min": 10, "str_max": 80,
+            "sound": "clang!",
+            "sound_fail": "ting.",
+            "ter_set": "t_floor",
```

Comment:
FWIW t_floor is wooden, last I knew.  Underground factory floor would probably be concrete, maybe rock.  (Doesn't burn if folks break out the flamers.)